src/backend/commands/matview.c

```diff
@@ -602,6 +588,15 @@ transientrel_init(QueryDesc *queryDesc)
 	heap_close(matviewRel, NoLock);
 }
 
+void
+transientrel_init(QueryDesc *queryDesc)
+{
+	if (transientrel_init_hook)
+		return (*transientrel_init_hook)(queryDesc);
+
+	transientrel_init_internal(queryDesc);
```

Comment:
Should we create a new file based DestReceiver for QE so that we don't need to hack transientrel_init here?

Reply:
We can add new DestReceiver, but we also need to hook cloudberry to use this new DestReceiver.